### ISSUE TITLE
mssql is not working 

### ISSUE THREAD

**Issue by Himanshukumar2812** [microsoft/vscode-mssql#17901]
Type: <b>Bug</b>

not getting the options to connect the database, not even show the icon on the side bar for the connection please check the issue 

Extension version: 1.22.1
VS Code version: Code 1.89.1 (dc96b837cf6bb4af9cd736aa3af08cf8279f7685, 2024-05-07T05:13:33.891Z)
OS version: Windows_NT x64 10.0.19045
Modes:

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|11th Gen Intel(R) Core(TM) i7-1185G7 @ 3.00GHz (8 x 1805)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: enabled_on<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_graphite: disabled_off<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: enabled|
|Load (avg)|undefined|
|Memory (System)|15.39GB (6.50GB free)|
|Process Argv|--crash-reporter-id 975af454-5bea-47c2-8695-fee123d578da|
|Screen Reader|no|
|VM|0%|
</details><details>
<summary>A/B Experiments</summary>

```
vsliv368:30146709
vspor879:30202332
vspor708:30202333
vspor363:30204092
tftest:31042121
vscoreces:30445986
vscod805cf:30301675
binariesv615:30325510
vsaa593cf:30376535
py29gd2263:31024239
c4g48928:30535728
azure-dev_surveyone:30548225
2i9eh265:30646982
962ge761:30959799
pythongtdpath:30769146
welcomedialogc:30910334
pythonidxpt:30866567
pythonnoceb:30805159
asynctok:30898717
pythontestfixt:30902429
pythonregdiag2:30936856
pyreplss1:30897532
pythonmypyd1:30879173
pythoncet0:30885854
h48ei257:31000450
pythontbext0:30879054
accentitlementsc:30995553
dsvsc016:30899300
dsvsc017:30899301
dsvsc018:30899302
cppperfnew:31000557
dsvsc020:30976470
pythonait:31006305
jchc7451:30973076
dsvsc021:30996838
bdiig495:31013172
pythoncenvpt:31043157
a69g1124:31038041
dwnewjupytercf:31039676

```

</details>

**Comment by kburtram** [comment]
@Himanshukumar2812 thanks for reporting this issue. Unfortunately, I'm not able to repro this problem.  Please try to update to 1.23.0 extension to see if that resolves the issue.

Regarding, not being able to see the side bar icon, it may be hidden.  You can check by right-clicking in the action bar and verifying the SQL Server entry is checked.  Also, please check the Help->Developer Tools Console to see if there are any errors being raised by the extension preventing it from loading.

![image](https://github.com/user-attachments/assets/2c1d87e1-2c20-46a3-9c7a-bd35f3effc4c)

**Comment by danielwestwoodABUHB** [comment]
I've had the same issue, to the point of removing VS code and reinstall. any updates welcomed - major issue

**Comment by kburtram** [comment]
@danielwestwoodABUHB is there any error in the Help->Developer Tools Console that would provide some indication of what's occurring here?  Is the SQL Server entry in the context menu available when right-clicking in the action bar (like shown above)?

**Comment by danielwestwoodABUHB** [comment]
I will certainly take a look when next online, you will have to forgive my inexperience in using the dev mode.

**Comment by kburtram** [comment]
Thanks @danielwestwoodABUHB!  For reference, there is a menu item in the Help menu named "Toggle Developer Tools".  That launches a diagnostic tool that has a Console tab (screenshot below).  Often when something isn't working correctly an exception call stack is output into this log that can help determine what error is occurring.

![image](https://github.com/user-attachments/assets/ab8321dc-a6c4-4636-838f-7220a1516f0f)

**Comment by kburtram** [comment]
For expired tokens, you could try to clear the token cache using the `Clear Microsoft Entra account token cache` command.  @cssuh any ideas why the extension isn't handling this scenario better?  e.g. asking for a token refresh rather than raising an exception.

![image](https://github.com/user-attachments/assets/a30aff69-0e7d-4b3f-82e1-a3df20a8b4c9)

**Comment by danielwestwoodABUHB** [comment]
No idea. Sorry - I’ve undertaken the clear command with no effect

Sent from Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by cssuh** [comment]
Hi @danielwestwoodABUHB, can you please try running this command as well? 
<img width="603" alt="Screenshot 2024-08-27 at 4 35 23 PM" src="https://github.com/user-attachments/assets/a10c93ae-5ac8-4657-b05c-2a545f95831e">

Also, can you confirm that you are running version 1.23.0?

**Comment by danielwestwoodABUHB** [comment]
Hello @cssuh - yes, updated, but get this error:

mssql: An error occurred while removing Microsoft Entra account: Cannot read properties of undefined (reading 'id')

**Comment by cssuh** [comment]
@danielwestwoodABUHB @kburtram I think what is happening here is the account object, which is stored in VSCode's [extensionContext.globalState](https://vscode-api.js.org/interfaces/vscode.ExtensionContext.html#globalState) is in some kind of bad state where it doesn't have a key.  I was able to generate the same error by setting my account key to undefined and attempting to remove it.  I've issued #18016 to fix this problem.
